test/cpp/qps/json_run_localhost.cc

```diff
@@ -112,20 +114,25 @@ int main(int argc, char** argv) {
     args.push_back(argv[i]);
   }
 
-  g_driver.reset(new SubProcess(args));
+  g_driver = new SubProcess(args);
   const int driver_join_status = g_driver->Join();
   if (driver_join_status != 0) {
     LogStatus(driver_join_status, "driver");
   }
-  for (const auto& worker : g_workers) {
-    if (worker) worker->Interrupt();
+  for (int i = 0; i < kNumWorkers; ++i) {
+    if (g_workers[i]) g_workers[i]->Interrupt();
   }
-  for (const auto& worker : g_workers) {
-    if (worker) {
-      const int worker_status = worker->Join();
+
+  for (int i = 0; i < kNumWorkers; ++i) {
+    if (g_workers[i]) {
+      const int worker_status = g_workers[i]->Join();
       if (worker_status != 0) {
         LogStatus(worker_status, "worker");
       }
     }
   }
+
+  delete g_driver;
```

Comment:
Do you want to do these two lines of deletion in the signal handler as well, just to be tidy?